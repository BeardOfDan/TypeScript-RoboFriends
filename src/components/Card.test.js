import { shallow } from 'enzyme';

import React from 'react';
import Card from './Card';

describe('Card component', () => {
  it('should render', () => {
    const testRobot = { name: 'testName', email: 'testEmail', id: 0 };
    expect(shallow(<Card robot={testRobot} />).length).toEqual(1);
    expect(shallow(<Card robot={testRobot} />)).toMatchSnapshot();
  });
});
