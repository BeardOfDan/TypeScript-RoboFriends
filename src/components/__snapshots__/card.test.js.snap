// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    robot={
        Object {
            "email": "testEmail",
            "id": 0,
            "name": "testName",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="testName' image"
          src="https://robohash.org/0?size=200x200"
/>,
        <div>
          <h2>
                    testName
          </h2>
          <p>
                    testEmail
          </p>
</div>,
      ],
      "className": "tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "testName' image",
          "src": "https://robohash.org/0?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              testName
</h2>,
            <p>
              testEmail
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testName",
            },
            "ref": null,
            "rendered": "testName",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testEmail",
            },
            "ref": null,
            "rendered": "testEmail",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="testName' image"
            src="https://robohash.org/0?size=200x200"
/>,
          <div>
            <h2>
                        testName
            </h2>
            <p>
                        testEmail
            </p>
</div>,
        ],
        "className": "tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "testName' image",
            "src": "https://robohash.org/0?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                testName
</h2>,
              <p>
                testEmail
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testName",
              },
              "ref": null,
              "rendered": "testName",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testEmail",
              },
              "ref": null,
              "rendered": "testEmail",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
